jsonschema-core.xml

```diff
@@ -363,15 +398,26 @@
                     This value SHOULD be normalized, and SHOULD NOT be an empty fragment &lt;#&gt; or an empty string &lt;&gt;.
                 </t>
                 <t>
-                    The root schema of a JSON Schema document SHOULD contain an "$id" keyword with an absolute-URI (containing a scheme, but no fragment).
+                    The root schema of a JSON Schema document SHOULD contain an "$id" keyword with 
+                    a URI (containing a scheme).  This URI SHOULD either not have a fragment, or
+                    have one that is an empty string.
+                    <!-- All of the standard meta-schemas use an empty fragment in their id/$id values. -->
+                    <cref>
+                        How should an "$id" URI reference containing a fragement with other components
+                        be interpreted?  There are two cases:  when the other components match
+                        the current base URI and when they change the base URI.
+                    </cref>
                 </t>
                 <t>
                     To name subschemas in a JSON Schema document,
                     subschemas can use "$id" to give themselves a document-local identifier.
-                    This form of "$id" keyword MUST begin with a hash ("#") to identify it as a fragment URI reference,
-                    followed by a letter ([A-Za-z]), followed by any number of
-                    letters, digits ([0-9]), hyphens ("-"), underscores ("_"), colons (":"), or periods (".").
-                    <!-- This restriction is the same one defined by XML -->
+                    This is done by setting "$id" to a URI reference consisting only of a fragment.
+                    The fragment identifier MUST NOT match the syntax of a JSON Pointer fragment
+                    identifier.
+                </t>
+                <t>
+                    The effect of defining an "$id" that is appears to be a JSON Pointer that does not match
+                    the expected JSON Pointer for that schema is not defined.
```

Comment:
I'm wary of changing this section (on the definition of the $id keyword) to liberalize what's acceptable. The specification should positively define what's acceptable in a regular grammar, without excluding a referenced set/relying on a compliment.

Otherwise, it might be the case that if the current RFC for JSON Pointer is superseded, that changes the behavior of this specification in a possibly incompatible way. The current behavior is the same restriction found in other documents like XML (that's also legal in a 7-bit URI), so the restriction shouldn't come as much of a surprise.

If JSON gets a fragment identifier defined, I don't want to assume it looks anything like JSON Pointer. I suspect it might operate like e.g. text/plain (defined in [RFC5147](https://tools.ietf.org/html/rfc5147) where it introduces a format like `#line=20` that's incompatible with e.g. XML's Name production.

Otherwise I like this PR. I'm trying to come up with some other comments, but this is turning into me re-reading XML and HTML.